src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataBlockService.cs

```diff
@@ -218,7 +246,9 @@ private async Task<DependentDataBlock> CreateDependentDataBlock(DataBlock block)
                 {
                     Id = f.Id,
                     Filename = f.Filename
-                }).ToList()
+                }).ToList(),
+                IsKeyStatistic = await _context.KeyStatisticsDataBlock
```

Comment:
Rather than hitting the KeyStatisticsDataBlock table with 2 queries per DataBlock that's getting created here, could we fetch all of the KeyStatisticsDataBlock DataBlockIds for this Release, *outside* of the `SelectAsync` loop, and then work out the `IsKeyStatistic` and `InContent` flags from there?

Reply:
```
                    var keyStatDataBlockIds = _context.KeyStatisticsDataBlock
                        .Where(ks => ks.ReleaseId == release.Id)
                        .Select(ks => ks.DataBlockId)
                        .ToList();
```

Reply:
Discussed this with Dunc. There is a possible performance improvement here, but it'll hurt readability to do this, so as performance isn't probably a big factor in this case, we'll leave it.